plasmapy/particles/particle_class.py

```diff
@@ -2053,30 +2055,46 @@ class CustomParticle(AbstractPhysicalParticle):
     >>> custom_particle = CustomParticle(
     ...     mass=1.2e-26 * u.kg,
     ...     charge=9.2e-19 * u.C,
-    ...     symbol="Ξ",
     ... )
     >>> custom_particle.mass
     <Quantity 1.2e-26 kg>
     >>> custom_particle.charge
     <Quantity 9.2e-19 C>
-    >>> custom_particle.symbol
-    'Ξ'
-    >>> import pytest
-    >>> with pytest.warns(UserWarning):
-    ...    custom_particle = CustomParticle(
-    ...        mass=1.5e-26 * u.kg,
-    ...        charge=-1,
-    ...        symbol="Ξ",
-    ...    )
-    >>> custom_particle.mass
+    >>> average_particle = CustomParticle(
+    ...     mass=1.5e-26 * u.kg,
+    ...     Z = -1.5,
+    ...     symbol="Ξ",
+    ... )
+    >>> average_particle.mass
     <Quantity 1.5e-26 kg>
-    >>> custom_particle.charge
-    <Quantity -1.60217...e-19 C>
-    >>> custom_particle.symbol
+    >>> average_particle.charge
+    <Quantity -2.40326...e-19 C>
+    >>> average_particle.symbol
     'Ξ'
     """
 
-    def __init__(self, mass: u.kg = None, charge: (u.C, Real) = None, symbol=None):
+    def __init__(
+        self,
+        mass: u.kg = None,
+        charge: u.C = None,
+        symbol: Optional[str] = None,
+        *,
+        Z: Optional[Real] = None,
+    ):
+
+        if isinstance(charge, Real):
+            warnings.warn(
+                "Providing a real number to 'charge' is deprecated. Use "
+                "'Z' as a keyword argument instead.",
```

Comment:
```suggestion
                "Providing a real number to 'charge' is deprecated. To specify the charge as a multiple of the elementary charge, use "
                "'Z' as a keyword argument instead, or pass e.g. `2 * u.e` into `charge`.",
```
will ofc need a pre-commit autofix for line length but you get the drift. Makes things more explicit for how we want people to use this.

Reply:
Sounds good!  I made a slight change of `.si` for the elementary charge because of how `astropy.units` handles electromagnetism.

Reply:
Oh, that thing! I always forget about it. Thanks!